src/ui/animation.js

```diff
@@ -45,50 +74,26 @@ export class Animation extends Phaser.Scene {
 
         // Moving xp text
         const xpText = this.add
-            .text(startX, startY, xp + " xp", font)
+            .text(xpStartX, startY, totalXp + " xp", font)
             .setOrigin(0.5, 0.5);
 
-        // Fixed skill text
-        const skillIcon = this.add
-            .image(startX, 25, skill + "Icon")
-            .setOrigin(0.5, 0.5)
-            .setDepth(1);
-
-        // Skill background
-        const shape = new Phaser.Geom.Circle(startX, 25, 20);
-        const graphics = this.add
-            .graphics({
-                lineStyle: { width: 1, color: 0x000000 },
-                fillStyle: { color: 0x707070 },
-            })
-            .fillCircleShape(shape)
-            .strokeCircleShape(shape);
+        targets.push(xpText);
 
         this.tweens.add({
-            targets: xpText,
-            x: endX,
+            targets: targets,
             y: endY,
             duration: 1000,
             ease: (t) => {
                 return Math.pow(Math.sin(t * 3), 3);
             },
             onComplete: () => {
-                xpText.destroy();
-                skillIcon.destroy();
-                graphics.destroy();
+                targets.forEach((target) => target.destroy());
             },
             onUpdate: () => {
                 // Destroy if within 1 pixel of end point
                 // Otherwise image will return to origin
-                if (
-                    xpText.x >= endX - 1 &&
-                    xpText.x <= endX + 1 &&
-                    xpText.y >= endY - 1 &&
-                    xpText.y <= endY + 1
-                ) {
-                    xpText.destroy();
-                    skillIcon.destroy();
-                    graphics.destroy();
+                if (xpText.y >= endY - 1 && xpText.y <= endY + 1) {
```

Comment:
Does removing the X axis check not mess up attack animations?

Reply:
This is only for the xp animation